blog/_posts/2020-10-27-tcp-mss-clamp.md

```diff
@@ -0,0 +1,63 @@
+---
+layout: post
+title: "TCPMSS Clamping in Firewalld"
+section: Blog
+date: 2020-10-28T21:52:00
+author: Vrinda Punj
+category: Feature
+---
+## What is TCPMSS Clamping?
+The maximum segment size is defined as the largest amount of data that can be received in a single
+TCP segment. TCPMSS clamping is a feature that sets the maximum segment size used by a TCP
+session. The way that it achieves this is during the TCP 3 way handshake, a server can set
+the MSS in the outgoing TCP SYN packets signalling the maximum segment
+size of the data packets that it can receive.
+
+## Why is it needed?
+The way that TCPMSS clamping was enabled in Firewalld before the addition of this feature
+was that users could add it using direct rules. An example of enabling the TCPMSS clamp
+feature through direct iptables rules would be the following command:
+
+    # firewall-cmd --permanent --direct --add-passthrough ipv4 -t mangle -I FORWARD -p tcp --syn -j TCPMSS --clamp-mss-to-pmtu
+
+In the example above, TCPMSS clamping is directly enabled by writing iptables rules.
+However, since Firewalld is supposed to be an abstraction of iptables and nftables, it is more clean
+to have it enabled as an option within Firewalld instead of having the user enable it by
+having to write direct rules.
+
+## What does this feature do?
+This feature adds TCPMSS clamping as an option in Firewalld rich rules, which gets
+translated into the corresponding rules for whichever backend is enabled in firewalld.conf.
+
+## What does this feature look like?
+This feature adds an enable TCPMSS clamp option to Firewalld rich rules. The user has an
+option called tcp-mss-clamp in rich rules. The tcp-mss-clamp option takes in an optional operand
```

Comment:
For anything that is syntax you should wrap it in backticks.
e.g.
tcp-mss-clamp vs `tcp-mss-clamp`
or
value vs `value`